lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -123,17 +134,36 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
         is_staff = bool(has_access(request.user, 'staff', course_key))
 
-        _, request.user = setup_masquerade(
-            request,
-            course_key,
-            staff_access=is_staff,
-            reset_masquerade_data=True
-        )
-
-        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        if student_id is None or student_id == request.user.id:
+            _, student = setup_masquerade(
+                request,
+                course_key,
+                staff_access=is_staff,
+                reset_masquerade_data=True
+            )
+        else:
+            # When a student_id is passed in, we display the progress page for the user
+            # with the provided user id, rather than the requesting user
+            try:
+                coach_access = has_ccx_coach_role(request.user, course_key)
+            except CCXLocatorValidationException:
+                coach_access = False
+
+            has_access_on_students_profiles = is_staff or coach_access
+            # Requesting access to a different student's profile
+            if not has_access_on_students_profiles:
+                raise Http404
+            try:
+                student = User.objects.get(id=student_id)
+            except User.DoesNotExist:
+                raise Http404
+
+        username = get_enterprise_learner_generic_name(request) or student.username
```

Comment:
This is interesting. This wasn't in old code - this actually looks like the first usage inside edx-platform? What's it's deal? Docstring says it worries about exposing sensitive SSO names. But feels weird that this is the only place we'd use it.

Reply:
Oh wow, I must have searched only *.py files - thanks, missed that. OK, I guess that's good then.

Might maybe be possibly good to indicate in the serialization that this is not strictly a username, but something good for display? Like `user_display_username` or something? But as I write that out, I hate it. `username` is fine.